### PR TITLE
Lower machine target for more compatibility

### PR DESCRIPTION
**Pull request by white238**
I will test this in a bit with a new devtools install that will work on ivybridge and broadwell chips

### CODE REVIEW

**Comment by jamiebramwell** [comment]
Should I merge this in yet?

**Comment by white238** [comment]
This does a lot more than previous and will need a proper review.  A lot of this work was moved over from my previous work on Axom. Here is a log of what it does now:

* Change to new version of finding devtools
* Fix config-build.py's ability to use command line build path
* Add script to build tpls for all specs on current machine locally and to shared area
* Add host-configs that use shared libs

**Comment by white238** [comment]
Note the gcc@8.1.0 host-config does not pass tests due to extra free()'s that if i remember correctly @samuelpmishLLNL is fixing in another PR.

**Comment by white238** [comment]
It also needs some documentation changes in the README.md but I've run out of steam at this point.

**Comment by white238** [comment]
This also bumps the BLT version to enable ClangFormat support

**Comment by jamiebramwell** [comment]
> 
> 
> Note the gcc@8.1.0 host-config does not pass tests due to extra free()'s that if i remember correctly @samuelpmishLLNL is fixing in another PR.

That's correct. I wouldn't worry about it for now.

File: scripts/uberenv/devtools.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (jamiebramwell) [inline scripts/uberenv/devtools.json]: Am I reading this right that uberenv now directly supports devtools?

File: scripts/uberenv/packages/serac/package.py [hunk 2]
```diff
@@ -72,12 +79,22 @@ class Serac(CMakePackage):
     variant('debug', default=False,
             description='Enable runtime safety and debug checks')
 
+    variant("devtools",  default=False,
```
Comment (jamiebramwell) [inline scripts/uberenv/packages/serac/package.py]: ahhh devtools is now a variant. I like it!

File: scripts/uberenv/spack_configs/toss_3_x86_64_ib/packages.yaml [hunk 3]
```diff
@@ -1,22 +1,16 @@
 packages:
   all:
+    # This defaults us to machine specific flags of ivybridge which allows
```
Comment (jamiebramwell) [inline scripts/uberenv/spack_configs/toss_3_x86_64_ib/packages.yaml]: Did you check if this works on rzalastor? I assume yes.

File: scripts/uberenv/devtools.json (hunk 1 above)
Comment (white238) [reply scripts/uberenv/devtools.json]: Uberenv supports it based on giving it another uberenv project file.  This PR ties the devtools together by pointing the serac package at the previously built devtools in the packages.yaml file.  It also allows you to build the whole TPL+devtools in one fell swoop, though I dont recommend it.

File: scripts/uberenv/spack_configs/toss_3_x86_64_ib/packages.yaml (hunk 3 above)
Comment (white238) [reply scripts/uberenv/spack_configs/toss_3_x86_64_ib/packages.yaml]: I ran astyle --version and it didnt core dump, so thats a plus.

**Comment by white238** [comment]
Link to new docs:

https://serac.readthedocs.io/en/bugfix-white238-lowerspacktarget/

**Comment by jamiebramwell** [comment]
@white238  Let me know when you consider this done and we can merge it.

**Comment by adrienbernede** [comment]
https://serac.readthedocs.io/en/bugfix-white238-lowerspacktarget/#building-serac
Serac *uses* a CMake build system?

**Comment by adrienbernede** [comment]
https://serac.readthedocs.io/en/bugfix-white238-lowerspacktarget/#getting-serac
Remove space in `-- recursive`

**Comment by white238** [comment]
> https://serac.readthedocs.io/en/bugfix-white238-lowerspacktarget/#building-serac
> Serac _uses_ a CMake build system?

Clearly the only thing that matters in Serac is the build system =)

**Comment by white238** [comment]
It's ready to merge once people find all my typo's and horrid documentation wording.

**Comment by jamiebramwell** [comment]
> 
> 
> > https://serac.readthedocs.io/en/bugfix-white238-lowerspacktarget/#building-serac
> > Serac _uses_ a CMake build system?
> 
> Clearly the only thing that matters in Serac is the build system =)

You joke, but that's one of my major sales points!

**Comment by jamiebramwell** [comment]
> 
> 
> It's ready to merge once people find all my typo's and horrid documentation wording.

I can take a close look after this spud goes to bed.

**Comment by white238** [comment]
> > > https://serac.readthedocs.io/en/bugfix-white238-lowerspacktarget/#building-serac
> > > Serac _uses_ a CMake build system?
> > 
> > 
> > Clearly the only thing that matters in Serac is the build system =)
> 
> You joke, but that's one of my major sales points!

Okay lets try this without editing Jamie's comment......

Aren't we all just build systems in the end?

**Comment by adrienbernede** [comment]
Well, you couldn’t be more right... My wife being pregnant at the moment. But I’m the lazy builder in this story :)

**Comment by jamiebramwell** [comment]
> 
> 
> Well, you couldn’t be more right... My wife being pregnant at the moment. But I’m the lazy builder in this story :)

Congrats!!!

**Comment by white238** [comment]
> Well, you couldn’t be more right... My wife being pregnant at the moment. But I’m the lazy builder in this story :)

Hopefully she's not due in the next month or so... My wife says hospitals are adding really strict rules for birth situations right now.

**Comment by adrienbernede** [comment]
This may not be the place to go into the details, but no, she is due in August, I was "just" forbidden to attend the ultra-sound exam today.

**Comment by white238** [comment]
> This may not be the place to go into the details, but no, she is due in August, I was "just" forbidden to attend the ultra-sound exam today.

Bummer =/

**Pull request merged**